### PR TITLE
Change syntax to get NC data

### LINKED ISSUE #394: Update NCDatasets compats in environments
See discussion in https://github.com/Alexander-Barth/NCDatasets.jl/issues/233

### PR DESCRIPTION
**Pull request by charleskawczynski**
Closes #394.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

**Comment by Sbozzolo** [comment]
My understanding is that the interface was designed to be agnostic of the dataloader. Is this still the case with this change? E.g., would if work with HDF5 instead of NCDatasets?

**Comment by Sbozzolo** [comment]
Moreover, this comment may be important:

              While fixing up my code I noticed that Array(v_cf) is not equivalent to v_cf.var[:] even in the vector case, because Array(v_cf) does Missing and DateTime conversion etc. The equivalent would be Array(v_cf.var). 

_Originally posted by @mwlidar in https://github.com/Alexander-Barth/NCDatasets.jl/issues/233#issuecomment-1761043485_
            
I don't know if it applies to us.

**Comment by charleskawczynski** [comment]
> My understanding is that the interface was designed to be agnostic of the dataloader. Is this still the case with this change? E.g., would if work with HDF5 instead of NCDatasets?

These changes are orthogonal to RRTMGP, since it (thankfully) doesn't depend on NCDatasets. If we decide to use HDF5 on the user side (in ClimaAtmos), then RRTMGP doesn't need to know. We could change it in RRTMGP's test/perf environments, but I figured the simple thing for now is to make RRTMGP compatible with the latest NCDatasets version so that things don't get stale.

**Comment by charleskawczynski** [comment]
>  because Array(v_cf) does Missing and DateTime conversion etc.... I don't know if it applies to us.

Yeah, I don't think we need to worry about that, since this really depends on the data, and our NC data apparently doesn't have missing / datetime stuff.

**Comment by Sbozzolo** [comment]
> > My understanding is that the interface was designed to be agnostic of the dataloader. Is this still the case with this change? E.g., would if work with HDF5 instead of NCDatasets?
> 
> These changes are orthogonal to RRTMGP, since it (thankfully) doesn't depend on NCDatasets. If we decide to use HDF5 on the user side (in ClimaAtmos), then RRTMGP doesn't need to know. We could change it in RRTMGP's test/perf environments, but I figured the simple thing for now is to make RRTMGP compatible with the latest NCDatasets version so that things don't get stale.

I don't think the problems are completely orthogonal. 

Consider for example the lines
https://github.com/CliMA/RRTMGP.jl/blob/1c1cf39bc6be8d99b73e5a70ff42e7209bd5dd76/src/optics/LookUpTables.jl#L243-L247
Whatever `ds` is, `Array(ds["kmajor"])` has to return a four dimensional array of floats. Similarly for the other lines. So, there's a coupling between the reader and RRTMGP (which is why it broke when `NCDatasets` changed its API). My comment was more along the lines of "is this syntax unique to `NCDatasets`, or it agreed upon in the community"? If this syntax is unique to `NCDatasets`, then we are essentially boxing ourselves to using it. However, as I understand it, `NCDatasets` changed the APIs to be more compatible with the `DiskArray` package, so hopefully this change will give us more compatibility. It'd be nice to check if `Array(HDF5["var"])` implements the same behavior.

**Comment by charleskawczynski** [comment]
> Whatever `ds` is, `Array(ds["kmajor"])` has to return a four dimensional array of floats...

Ah, whoops, I didn't even realize I was making changes to src. I see. I haven't looked closely enough, but it would be nice if we could move this logic outside of RRTMGP.

> My comment was more along the lines of "is this syntax unique to `NCDatasets`, or it agreed upon in the community"? If this syntax is unique to `NCDatasets`, then we are essentially boxing ourselves to using it. However, as I understand it, `NCDatasets` changed the APIs to be more compatible with the `DiskArray` package, so hopefully this change will give us more compatibility. It'd be nice to check if `Array(HDF5["var"])` implements the same behavior.

Agreed.

**Comment by Sbozzolo** [comment]
> My comment was more along the lines of "is this syntax unique to NCDatasets, or it agreed upon in the community"? If this syntax is unique to NCDatasets, then we are essentially boxing ourselves to using it. However, as I understand it, NCDatasets changed the APIs to be more compatible with the DiskArray package, so hopefully this change will give us more compatibility. It'd be nice to check if Array(HDF5["var"]) implements the same behavior.
> 
> Agreed.

I checked with `HDF5.jl` and found that `Array(ds["var"])` returns a multidimensional array as in `NCDatasets`.

**Review by Sbozzolo — APPROVED** [review]

File: examples/Project.toml [hunk 1]
```diff
@@ -21,4 +21,4 @@ Suppressor = "fd094767-a336-5f1f-9728-57cf17d0bbfb"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [compat]
-NCDatasets = "=0.12.17"
+NCDatasets = "0.12.17, 0.13"
```
Comment (Sbozzolo) [inline examples/Project.toml:24]: Is this new syntax compatible with NCDatasets 0.12.17 ?

Maybe we should just require > 0.13.

**Review by Sbozzolo — APPROVED** [review]

**Comment by Sbozzolo** [comment]
Any reason not to merge this?

**Comment by charleskawczynski** [comment]
No, i think we can merge

**Comment by Sbozzolo** [comment]
Buildkite didn't start, so we probably haven't set it right. Let me look at that

**Comment by Sbozzolo** [comment]
Okay, it was filtering the branches. Now it works. I also bumped the version so that we can release the fixed version.

**Pull request merged**